### ISSUE TITLE
--help all output is sometimes cut

### ISSUE THREAD

**Issue by kuba** [certbot/certbot#1066]
User reported that sometimes `--help all` cuts the head of the output:

```
root@manager ~/letsencrypt # ./letsencrypt-auto --help all
Updating letsencrypt and virtual environment dependencies......
Running with virtualenv: /root/.local/share/letsencrypt/bin/letsencrypt --help all
usage: 
  letsencrypt [SUBCOMMAND] [options] [-d domain] [-d domain] ...

apache:
  Apache Web Server - Alpha

  --apache-ctl APACHE_CTL
                        Path to the 'apache2ctl' binary, used for
                        'configtest', retrieving the Apache2 version number,
                        and initialization parameters. (default: apache2ctl)
  --apache-enmod APACHE_ENMOD
                        Path to the Apache 'a2enmod' binary. (default:
                        a2enmod)
  --apache-dismod APACHE_DISMOD
                        Path to the Apache 'a2enmod' binary. (default:
                        a2dismod)
  --apache-init-script APACHE_INIT_SCRIPT
                        Path to the Apache init script (used for server
                        reload/restart). (default: /etc/init.d/apache2)
  --apache-le-vhost-ext APACHE_LE_VHOST_EXT
                        SSL vhost configuration extension. (default: -le-
                        ssl.conf)
  --apache-server-root APACHE_SERVER_ROOT
                        Apache server root directory. (default: /etc/apache2)

webroot:
  Webroot Authenticator

  --webroot-path WEBROOT_PATH
                        public_html / webroot path (default: None)

null:
  Null Installer

manual:
  Manual Authenticator

  --manual-test-mode    Test mode. Executes the manual command in subprocess.
                        (default: False)

standalone:
  Standalone Authenticator

  --standalone-supported-challenges STANDALONE_SUPPORTED_CHALLENGES
                        Supported challenges, order preferences are randomly
                        chosen. (default: dvsni,simpleHttp)

auth:
  Options for modifying how a cert is obtained

  --csr CSR             Path to a Certificate Signing Request (CSR) in DER
                        format. (default: None)

install:
  Options for modifying how a cert is deployed

revoke:
  Options for revocation of certs

rollback:
  Options for reverting config changes

  --checkpoints N       Revert configuration N number of checkpoints.
                        (default: 1)

plugins:
  Plugin options

  --init                Initialize plugins. (default: False)
  --prepare             Initialize and prepare plugins. (default: False)
  --authenticators      Limit to authenticator plugins only. (default: None)
  --installers          Limit to installer plugins only. (default: None)
```

Screenshot below starts at the "apache" section...

![letsencrypt2](https://cloud.githubusercontent.com/assets/106176/10645587/089c70e0-781e-11e5-8f1c-b0d2f8cee4ec.png)

**Comment by kuba** [comment]
Another screenshot from the same terminal session:
![letsencrypt](https://cloud.githubusercontent.com/assets/106176/10645669/7bbd670a-781e-11e5-9690-2fafeb262683.png)

**Comment by pde** [comment]
@kuba Can you point me to the user who reported this?

**Comment by pde** [comment]
Initial questions: what OS, version and hosting provider are you running this on?

**Comment by ArchimedesPi** [comment]
@kuba IIRC a person on IRC was asking you about this. Do you have the logs of that conversation?

**Comment by pde** [comment]
Since this seems to be unreproducible and rare, kicking it out of the launch blocker list for now.

**Comment by ArchimedesPi** [comment]
@kuba @pde [IRC logs of the bug report](https://gist.github.com/ArchimedesPi/a402a695d8b4047d82a7)

tl;dr:
- debian jessie
- KDE, Konsole
- weird prompt coloration
- `ls` highlighting is on
- sometimes the full text is printed, sometimes not

IMHO it's probably in our wrapper layer around argparse. It's *very* unlikely that argparse has a bug like this given how widely it's used.

**Comment by pde** [comment]
@ArchimedesPi agree it could be our wrapper layer in the abstract.  It could also be a bug in `configargparse`, or a bug in the way we use argparse.  A few things to try, if anyone has a clean reproduction of this:

* Does replacing `configargparse.ArgParser` with `argparse.ArgumentParser` fix the problem? (Obviously that won't work for the client in general, but it should probably be possible with `--help`)
* Does removing `formatter_class=argparse.ArgumentDefaultsHelpFormatter` fix the problem?
* Does this ever happen with help on more specific topics?  Or only with `--help all`?

**Comment by SwartzCr** [comment]
closing as unreproduced
